### PR TITLE
feat(cluster): add deployment error banner

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-847

You can test with this [Cluster](http://localhost:4200/organization/460616f0-94da-4d35-b631-6fa4ed08eb9a/cluster/be9e22b0-d05a-4330-b5b5-547667d380fd/settings/general) 

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by acarranoqovery** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery  preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery  preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Programs must be written for people to read, and only incidentally for machines to execute

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] my-env - feat(cluster): add deployment error banner - 2023-10-26T16:04:36Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/c83b7df2-e605-4bfe-9759-754f5590d246/services/general)

 Another comment will be posted when deployments are terminated

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [\[PR\] my-env - feat(cluster): add deployment error banner - 2023-10-26T16:04:36Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/e773b890-e6b2-4406-93a9-71f9a7ab6d73/environment/c83b7df2-e605-4bfe-9759-754f5590d246/logs)

**Review by ctjhoa — CHANGES_REQUESTED** [review]

File: libs/pages/layout/src/lib/ui/layout-page/layout-page.tsx [hunk 1]
```diff
@@ -12,29 +17,37 @@ export interface LayoutPageProps {
...
-export const displayClusterBanner = (status?: ClusterStateEnum): boolean => {
+export const displayClusterDeploymentBanner = (status?: ClusterStateEnum): boolean => {
   return match(status)
     .with(ClusterStateEnum.DEPLOYMENT_QUEUED, ClusterStateEnum.DEPLOYING, () => true)
     .otherwise(() => false)
 }
 
+export const displayClusterCredentialErrorBanner = (clusters?: ClusterEntity[]): ClusterEntity[] => {
+  if (!clusters) return []
+  const clustersDeploymentError = clusters?.filter((c) => c.status === ClusterStateEnum.DEPLOYMENT_ERROR)
```
[the remark this answers is not in the document]
Comment (RemiBonnet) [reply libs/pages/layout/src/lib/ui/layout-page/layout-page.tsx]: https://github.com/Qovery/console/pull/923/commits/e345d07c1f873a9021a0c9246c370ae92e6cf5c3

**Review by ctjhoa — APPROVED** [review]

**Pull request merged**